warehouse/admin.go

```diff
@@ -77,30 +88,34 @@ func (wh *WarehouseAdmin) Query(s QueryInput, reply *warehouseutils.QueryResult)
 	return err
 }
 
-func (wh *WarehouseAdmin) QueryWhUploads(uploadsReq UploadsReqT, reply *[]byte) error {
-	uploadsReq.API = UploadAPI
-	res, err := uploadsReq.GetWhUploads()
-	if err != nil {
-		return err
+// Test the underlying warehouse
+func (wh *WarehouseAdmin) Test(s TestInput, reply *TestResponse) (err error) {
+	if strings.TrimSpace(s.DestID) == "" {
+		return errors.New("please specify the destination ID to query the warehouse")
 	}
-	bytes, err := json.Marshal(res)
-	if err != nil {
-		return err
+
+	var warehouse warehouseutils.WarehouseT
+	srcMap, ok := connectionsMap[s.DestID]
+	if !ok {
+		return errors.New("please specify a valid and existing destination ID")
```

Comment:
Add the destinationID in the error response, so we can log it appropriately. Wdyt ?